blockstore/apps/api/v1/views/bundles.py

```diff
@@ -0,0 +1,52 @@
+"""
+Views for Bundles and BundleVersions.
+"""
+
+from rest_framework import viewsets
+from rest_framework.generics import get_object_or_404
+
+from blockstore.apps.bundles.models import Bundle, BundleVersion
+
+from ...constants import UUID4_REGEX, VERSION_NUM_REGEX
+from ..serializers.bundles import BundleSerializer, BundleVersionSerializer
+
+
+class BundleViewSet(viewsets.ModelViewSet):
+    """
+    ViewSet for Bundle model.
+    """
+
+    lookup_field = 'uuid'
+    lookup_url_kwarg = 'bundle_uuid'
+    lookup_value_regex = UUID4_REGEX
+
+    queryset = Bundle.objects.all()
+    serializer_class = BundleSerializer
+
+
+class BundleVersionViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    ViewSet for BundleVersion model.
+    """
+
+    lookup_fields = ('bundle__uuid', 'version_num')
+    lookup_url_kwargs = ('bundle_uuid', 'version_num')
+    lookup_value_regexes = (UUID4_REGEX, VERSION_NUM_REGEX)
```

Comment:
I couldn't figure out how to get a list of all `BundleVersion` objects for a given `Bundle` -- both the `bundle_uuid` and the `version_num` must be in the path, else there's a 404.

This would be useful -- is it possible?

Reply:
`<bundle_uuid>,<version_num>` is a composite identifier. Its like a `usage_key` incorporates the `course_key`, `block_type` and `uuid`. To be able to list `BundleVersions` for a `Bundle`, we will need a filter `/api/v1/bundle_versions/?bundle_uuid=<bundle_uuid>`.

There are some details related to it under "Identifying Resources" in the [edX REST API Conventions](https://openedx.atlassian.net/wiki/spaces/OpenDev/pages/18350757/edX+REST+API+Conventions).

Todo for filters has been added to LX-53.

Reply:
👍